homeassistant/components/fritz/camera.py

```diff
@@ -0,0 +1,115 @@
+"""FRITZ camera integration.
+
+Currently only used to display QR codes.
+"""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+import logging
+
+from homeassistant.components.camera import Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util, slugify
+
+from .common import AvmWrapper, FritzBoxBaseEntity
+from .const import DEFAULT_GUEST_WIFI_QR_REFRESH_SEC, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add a entities from a config_entry."""
+    _LOGGER.debug("Setting up FRITZ!Box camera entities")
+    avm_wrapper: AvmWrapper = hass.data[DOMAIN][entry.entry_id]
+
+    entities = await hass.async_add_executor_job(
+        _setup_guest_wifi_qr, avm_wrapper, entry.title
+    )
+
+    async_add_entities(entities, True)
+
+
+def _setup_guest_wifi_qr(
+    avm_wrapper: AvmWrapper, device_friendly_name: str
+) -> list[FritzGuestWifiQRCamera]:
+    """Set up guest wifi entity to display the qr code."""
+    _LOGGER.debug("Setting up qr code camera entity")
+    ssid = avm_wrapper.fritz_guest_wifi.ssid
+    return [
+        FritzGuestWifiQRCamera(
+            avm_wrapper,
+            device_friendly_name,
+            ssid,
+        )
+    ]
+
+
+class FritzGuestWifiQRCamera(FritzBoxBaseEntity, Camera):
+    """Implementation of the FritzBox guest wifi QR code camera entity."""
+
+    QR_CODE = "QR code"
+
+    def __init__(
+        self,
+        avm_wrapper: AvmWrapper,
+        device_friendly_name: str,
+        ssid: str,
+    ) -> None:
+        """Initialize the camera."""
+        Camera.__init__(self)
+
+        self._attr_name = f"{device_friendly_name} {ssid} {self.QR_CODE}"
+        self._attr_unique_id = (
+            f"{avm_wrapper.unique_id}-guest-wlan-{slugify(self.QR_CODE)}"
```

Comment:
Why do we need to add the static suffix to the unique id? Are we planning to add another camera entity in the future?

Reply:
Device has the ability to generate the QR code also for main WLAN setting.
I'm not sure it makes sense to add it, but to avoid any potential conflict/breaking change in the future I think we should consider the static suffix.

Simone

Reply:
Ok.